### DISCUSSION TITLE
Injeção de dependência

### DISCUSSION THREAD

**Discussion by RanieryCG** [balta-io/7194#23] [Perguntas e Respostas] [answered]
Quando é falado sobre injeção de dependência o certo não seria no construtor do controller, como na documentação da Microsoft? 
https://docs.microsoft.com/pt-br/aspnet/core/mvc/controllers/dependency-injection?view=aspnetcore-5.0

Qual a diferença de passar no controller ou usar o AddScoped?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @RanieryCG, DI é uma técnica, que pode ser utilizada tanto no construtor quanto via `[FromServices]` no ASP.NET.

Estes artigos vão te ajudar:
- https://balta.io/blog/dependency-injection
- https://balta.io/blog/aspnet-core-dependency-injection

E este vídeo também:
https://www.youtube.com/watch?v=va9wX68lAfA

**Reply by RanieryCG** [reply]
Obrigado, @andrebaltieri! Vou olhar!
